### ISSUE TITLE
sophomorix-print does not work anymore: TeX capacity exceeded, sorry [input stack size=5000]

### ISSUE THREAD

**Issue by zefanja** [linuxmuster/sophomorix4#13]
The latest version can't print all users anymore, because of a TeX error: `TeX capacity exceeded, sorry [input stack size=5000]`

Here is the log:

```
root@server:~# sophomorix-print -vv
All options  were recognized.
{}
{
  "CONFIG" => {
    "ACTION" => {},
    "DEPENDS" => {},
    "MAYBE" => {},
    "ONE_OF" => {},
    "SOME_OF" => {}
  },
  "CONFIGURED" => {
    "help" => "TRUE",
    "info" => "TRUE",
    "json" => "TRUE",
    "verbose" => "TRUE"
  },
  "MODIFIER_OPTIONS" => {
    "json" => "TRUE",
    "verbose" => "TRUE"
  },
  "info" => 0,
  "json" => 0,
  "verbose" => 3
}
Option verbose is a modifier option
Option json is a modifier option
Hmmh. do not know what to do with option info
* forcing info mode
Option combinations successfully checked
   Checking Samba4 AD connection ...
Retrieving RootDSE...
      * NamingContext: <DC=internal,DC=cdsc,DC=ac,DC=th>
      * NamingContext: <CN=Configuration,DC=internal,DC=cdsc,DC=ac,DC=th>
      * NamingContext: <CN=Schema,CN=Configuration,DC=internal,DC=cdsc,DC=ac,DC=th>
      * NamingContext: <DC=DomainDnsZones,DC=internal,DC=cdsc,DC=ac,DC=th>
      * NamingContext: <DC=ForestDnsZones,DC=internal,DC=cdsc,DC=ac,DC=th>
   * RootDSE: DC=internal,DC=cdsc,DC=ac,DC=th
Binding with CN=administrator,CN=Users,DC=internal,DC=cdsc,DC=ac,DC=th
Testing if the Sophomorix Schema exists (Sophomorix-User)...
   * Sophomorix-Schema exists
Sophomorix::SophomorixSambaAD::AD_dns_get RootDSE: DC=internal,DC=cdsc,DC=ac,DC=th -> DNS: internal.cdsc.ac.th

################################################################################
 # Reading /usr/share/sophomorix/devel/sophomorix.ini                     # 
################################################################################
Distro-check: Ubuntu 18.04 is OK

################################################################################
 # Reading /etc/samba/smb.conf                                            # 
################################################################################

################################################################################
 # Parsing: net conf list                                                 # 
################################################################################

################################################################################
 # Asking domain passwordsettings from samba                              # 
################################################################################
   * <Password_complexity> ---> <on>
   * <Store_plaintext_passwords> ---> <off>
   * <Password_history_length> ---> <24>
   * <Minimum_password_length> ---> <7>
   * <Minimum_password_age_days> ---> <1>
   * <Maximum_password_age_days> ---> <42>
   * <Account_lockout_duration_mins> ---> <30>
   * <Account_lockout_threshold_attempts> ---> <0>
   * <Reset_account_lockout_after_mins> ---> <30>

################################################################################
 # Reading /usr/share/sophomorix/devel/master/sophomorix.conf.master      # 
################################################################################

################################################################################
 # Reading /etc/linuxmuster/sophomorix/sophomorix.conf                    # 
################################################################################
   * global: LANG ---> <DE>
   * global: USE_QUOTA ---> <TRUE>
   * global: ADMINS_PRINT ---> <Placeholder for admin name>
   * global: SCHOOLS ---> <>

################################################################################
 # OK: default-school share exists                                        # 
################################################################################

################################################################################
 # OK: /etc/linuxmuster/sophomorix/default-school/school.conf             # 
################################################################################

################################################################################
 # Reading /usr/share/sophomorix/devel/master/school.conf.master          # 
################################################################################

################################################################################
 # Reading /etc/linuxmuster/sophomorix/default-school/school.conf         # 
################################################################################
   * SCHOOL default-school: Para: ADMINS_PRINT -> <The String that describes the admins>
   * SCHOOL default-school: Para: SCHOOL_LONGNAME -> <CDSC>
   * SCHOOL default-school: Para: MAILTYPE -> <none>
   * SCHOOL default-school: Para: LANG -> <EN>
   * ROLE student: QUOTA_DEFAULT_GLOBAL ---> <1006>
   * ROLE student: QUOTA_DEFAULT_SCHOOL ---> <1506>
   * ROLE student: MAILQUOTA_DEFAULT ---> <156>
   * ROLE student: CLOUDQUOTA_PERCENTAGE ---> <100>
   * ROLE student: WEBUI ---> <>
   * ROLE student: ANOTHER_UI ---> <>
   * ROLE teacher: QUOTA_DEFAULT_GLOBAL ---> <2006>
   * ROLE teacher: QUOTA_DEFAULT_SCHOOL ---> <2506>
   * ROLE teacher: MAILQUOTA_DEFAULT ---> <306>
   * ROLE teacher: CLOUDQUOTA_PERCENTAGE ---> <100>
   * ROLE teacher: WEBUI ---> <>
   * ROLE teacher: ANOTHER_UI ---> <>
   * ROLE schooladministrator: QUOTA_DEFAULT_GLOBAL ---> <3006>
   * ROLE schooladministrator: QUOTA_DEFAULT_SCHOOL ---> <3506>
   * ROLE schooladministrator: MAILQUOTA_DEFAULT ---> <506>
   * ROLE schooladministrator: CLOUDQUOTA_PERCENTAGE ---> <100>
   * ROLE schooladministrator: WEBUI ---> <>
   * ROLE schooladministrator: ANOTHER_UI ---> <>
   * FILE students.csv: FILTERSCRIPT ---> <--->
   * FILE students.csv: ENCODING ---> <auto>
   * FILE students.csv: ENCODING_FORCE ---> <FALSE>
   * FILE students.csv: SURNAME_CHARS ---> <6>
   * FILE students.csv: FIRSTNAME_CHARS ---> <2>
   * FILE students.csv: SURNAME_FIRSTNAME_REVERSE ---> <FALSE>
   * FILE students.csv: RANDOM_PWD ---> <TRUE>
   * FILE students.csv: PWD_LENGTH ---> <10>
   * FILE students.csv: TOLERATION_TIME ---> <30>
   * FILE students.csv: DEACTIVATION_TIME ---> <60>
   * FILE teachers.csv: FILTERSCRIPT ---> <--->
   * FILE teachers.csv: ENCODING ---> <auto>
   * FILE teachers.csv: ENCODING_FORCE ---> <FALSE>
   * FILE teachers.csv: SURNAME_CHARS ---> <0>
   * FILE teachers.csv: FIRSTNAME_CHARS ---> <0>
   * FILE teachers.csv: SURNAME_FIRSTNAME_REVERSE ---> <FALSE>
   * FILE teachers.csv: RANDOM_PWD ---> <TRUE>
   * FILE teachers.csv: PWD_LENGTH ---> <12>
   * FILE teachers.csv: TOLERATION_TIME ---> <60>
   * FILE teachers.csv: DEACTIVATION_TIME ---> <90>
   * FILE extrastudents.csv: FILTERSCRIPT ---> <--->
   * FILE extrastudents.csv: ENCODING ---> <auto>
   * FILE extrastudents.csv: ENCODING_FORCE ---> <FALSE>
   * FILE extrastudents.csv: SURNAME_CHARS ---> <6>
   * FILE extrastudents.csv: FIRSTNAME_CHARS ---> <2>
   * FILE extrastudents.csv: SURNAME_FIRSTNAME_REVERSE ---> <FALSE>
   * FILE extrastudents.csv: RANDOM_PWD ---> <TRUE>
   * FILE extrastudents.csv: PWD_LENGTH ---> <10>
   * FILE extrastudents.csv: TOLERATION_TIME ---> <40>
   * FILE extrastudents.csv: DEACTIVATION_TIME ---> <80>
   * FILE extraclasses.csv: PWD_LENGTH ---> <10>

################################################################################
 # /usr/sbin/sophomorix-print started ...                                 # 
################################################################################

################################################################################
 # Basename of output files: add-unknown                                  # 
################################################################################

################################################################################
 # Template dir:  /usr/share/sophomorix/lang/latex/templates              # 
################################################################################

################################################################################
 # Template file: datalist-EN-36-template.tex                             # 
################################################################################

################################################################################
 # Looking for printout data in school default-school                     # 
################################################################################

################################################################################
 # 4 sophomorix users found for password printout                         # 
################################################################################
   * Creating datablock for teachers

################################################################################
 # latex is creating add-unknown.dvi                                      # 
################################################################################
  * cd /var/lib/sophomorix/print-data; /usr/bin/latex add-unknown.tex 
This is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian) (preloaded format=latex)
 restricted \write18 enabled.
entering extended mode
(./add-unknown.tex
LaTeX2e <2017-04-15>
Babel <3.18> and hyphenation patterns for 8 language(s) loaded.
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/fleqn.clo)
(/usr/share/texlive/texmf-dist/tex/latex/base/size11.clo))
(/usr/share/texlive/texmf-dist/tex/latex/base/fontenc.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.def))
(/usr/share/texlive/texmf-dist/tex/latex/base/inputenc.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/utf8.def
(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.dfu)
(/usr/share/texlive/texmf-dist/tex/latex/base/ot1enc.dfu)
(/usr/share/texlive/texmf-dist/tex/latex/base/omsenc.dfu)))
(/usr/share/texlive/texmf-dist/tex/generic/german/ngerman.sty v2.5e 1998-07-08)
 (/usr/share/texlive/texmf-dist/tex/latex/graphics/color.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/graphics-def/dvips.def)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/dvipsnam.def))
(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.sty
(/usr/share/texlive/texmf-dist/tex/generic/babel/switch.def)
(/usr/share/texlive/texmf-dist/tex/generic/babel-english/english.ldf
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.def
(/usr/share/texlive/texmf-dist/tex/generic/babel/txtbabel.def))
! TeX capacity exceeded, sorry [input stack size=5000].
\mdqoff ->\catcode `\"1
                       2\relax 
l.47 \LdfInit\CurrentOption{date\CurrentOption}
                                               
No pages of output.
Transcript written on add-unknown.log.
This is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian) (preloaded format=latex)
 restricted \write18 enabled.
entering extended mode
(./add-unknown.tex
LaTeX2e <2017-04-15>
Babel <3.18> and hyphenation patterns for 8 language(s) loaded.
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/fleqn.clo)
(/usr/share/texlive/texmf-dist/tex/latex/base/size11.clo))
(/usr/share/texlive/texmf-dist/tex/latex/base/fontenc.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.def))
(/usr/share/texlive/texmf-dist/tex/latex/base/inputenc.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/utf8.def
(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.dfu)
(/usr/share/texlive/texmf-dist/tex/latex/base/ot1enc.dfu)
(/usr/share/texlive/texmf-dist/tex/latex/base/omsenc.dfu)))
(/usr/share/texlive/texmf-dist/tex/generic/german/ngerman.sty v2.5e 1998-07-08)
 (/usr/share/texlive/texmf-dist/tex/latex/graphics/color.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/graphics-def/dvips.def)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/dvipsnam.def))
(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.sty
(/usr/share/texlive/texmf-dist/tex/generic/babel/switch.def)
(/usr/share/texlive/texmf-dist/tex/generic/babel-english/english.ldf
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.def
(/usr/share/texlive/texmf-dist/tex/generic/babel/txtbabel.def))
! TeX capacity exceeded, sorry [input stack size=5000].
\mdqoff ->\catcode `\"1
                       2\relax 
l.47 \LdfInit\CurrentOption{date\CurrentOption}
                                               
No pages of output.
Transcript written on add-unknown.log.

################################################################################
 # dvips is creating add-unknown.ps                                       # 
################################################################################
  * cd /var/lib/sophomorix/print-data; /usr/bin/dvips add-unknown.dvi 
This is dvips(k) 5.997 Copyright 2017 Radical Eye Software (www.radicaleye.com)
/usr/bin/dvips: DVI file can't be opened: add-unknown.dvi: No such file or directory

################################################################################
 # ps2pdf is creating add-unknown.pdf                                     # 
################################################################################
  * cd /var/lib/sophomorix/print-data; /usr/bin/ps2pdf add-unknown.ps 

################################################################################
 # TEX-RESULT: /var/lib/sophomorix/print-data/add-unknown.tex             # 
################################################################################

################################################################################
 # PDF-RESULT: /var/lib/sophomorix/print-data/add-unknown.pdf             # 
################################################################################

################################################################################
 # /usr/sbin/sophomorix-print terminated regularly                        # 
################################################################################
Calling console printout
LOG       : /usr/sbin/sophomorix-print terminated regularly
```

**Comment by zefanja** [comment]
It's the same for the german template. If I remove / comment out `\usepackage{ngerman}` it does work again.

The templates for a single user are working as expected - both with DE and EN.

**Comment by jeffbeck** [comment]
Did you test with a lot of users?

**Comment by zefanja** [comment]
No, just 3 or 4.

**Comment by jeffbeck** [comment]
OK; I see the ERROR too, only on the 36 template

**Comment by jeffbeck** [comment]
Tex capacity is by page as I remember (memory is cleared after typesetting the page). 
Increasing Tex capacity should work for any number of pages.

**Comment by jeffbeck** [comment]
I tested this to increase main memory, but without success.

https://gordonlesti.com/debian-increase-latex-main-memory-size/

So the error must be connected to the \usepackage{babel} coll.

I will look into it tomorrow. On my ubuntu 14.04 Computer the template compiles without errors.

I make another package (without babel, but fixed issue #12 )

**Comment by jeffbeck** [comment]
I fixed it in V-3.42.1-3
babel ngerman is not allowed with ngerman
